### PR TITLE
zsh-abbr: add module

### PR DESCRIPTION
**Pull request by ilaumjd**
### Description

Add `zsh-abbr` module under `zsh`.
Add `zsh-abbr.abbrs` option so we can use fish-like abbreviations in zsh.

References: https://github.com/olets/zsh-abbr

### Checklist

- [x] Change is backwards compatible.

- [x] Code formatted with `./format`.

- [x] Code tested through `nix-shell --pure tests -A run.all` or `nix develop --ignore-environment .#all` using Flakes.

- [x] Test cases updated/added. See [example](https://github.com/nix-community/home-manager/commit/f3fbb50b68df20da47f9b0def5607857fcc0d021#diff-b61a6d542f9036550ba9c401c80f00ef).

- [x] Commit messages are formatted like

    ```
    {component}: {description}

    {long description}
    ```

    See [CONTRIBUTING](https://github.com/nix-community/home-manager/blob/master/docs/contributing.adoc#sec-commit-style) for more information and [recent commit messages](https://github.com/nix-community/home-manager/commits/master) for examples.

- If this PR adds a new module

  - [x] Added myself as module maintainer. See [example](https://github.com/nix-community/home-manager/blob/068ff76a10e95820f886ac46957edcff4e44621d/modules/programs/lesspipe.nix#L6).

### CODE REVIEW

File: modules/programs/zsh/zsh-abbr.nix [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+with lib;
+let cfg = config.programs.zsh.zsh-abbr;
+in {
+  meta.maintainers = [ hm.maintainers.ilaumjd ];
+
+  options.programs.zsh.zsh-abbr = {
+    enable =
+      mkEnableOption "zsh-abbr - zsh manager for auto-expanding abbreviations";
+
+    abbrs = mkOption {
```
Comment (rycee) [inline modules/programs/zsh/zsh-abbr.nix:12]: Maybe can expand the name for clarity?

```suggestion
    abbreviations = mkOption {
```

File: modules/programs/zsh/zsh-abbr.nix [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+      example = {
+        l = "less";
+        gco = "git checkout";
+      };
+      description = ''
+        An attribute set that maps aliases (the top level attribute names
+        in this option) to abbreviations. Abbreviations are expanded with
+        the longer phrase after they are entered.
+      '';
+      type = types.attrsOf types.str;
```
Comment (rycee) [inline modules/programs/zsh/zsh-abbr.nix:23]: The type should be the first attribute in the option, i.e., before the `default = { };` line.

File: modules/programs/zsh/zsh-abbr.nix [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    programs.zsh.plugins = [{
+      name = "zsh-abbr";
+      src = pkgs.zsh-abbr;
+      file = "/share/zsh/zsh-abbr/abbr.plugin.zsh";
+    }];
+
+    xdg.configFile = {
+      "zsh-abbr/user-abbreviations".text = concatStringsSep "\n"
+        (mapAttrsToList (k: v: "abbr ${escapeShellArg k}=${escapeShellArg v}")
+          cfg.abbrs);
```
Comment (rycee) [inline modules/programs/zsh/zsh-abbr.nix:37]: Typically it is nice to have a final newline
```suggestion
          cfg.abbrs) + "\n";
```

**Comment by rycee** [comment]
Thanks for the contribution! I've added a few minor comments.

**Comment by rycee** [comment]
Oh, and as you can see in the CI result, you need to add yourself to the `lib/maintainers.nix` file.

**Review by ilaumjd — COMMENTED** [review]
@rycee Thanks, pushed the changes.
For the maintainer i actually added on a different [PR](https://github.com/nix-community/home-manager/pull/4517), will delete it there if this is merged.

Also, these are my first PR to an open source, so thanks for the feedbacks :)

**Comment by rycee** [comment]
Ah, sorry. Missed the other module PR. I'll merge this first, thanks for the contribution looks very nice! 🙂

**Comment by rycee** [comment]
Oh, and congratulations on doing your first PR. I wish you luck with your future contributions!

**Pull request merged**